Sort-Object: more about default properties (#6375)
MicrosoftDocs/PowerShell-Docs: reference/7.1/Microsoft.PowerShell.Utility/Sort-Object.md

## DESCRIPTION

The `Sort-Object` cmdlet sorts objects in ascending or descending order based on object property

> values. If sort properties are not included in a command, PowerShell uses default sort properties,
> defined as the `DefaultKeyPropertySet` of the type data of the first input object,
> or compares the objects themselves.

Reviewer: ```suggestion
values. If sort properties are not included in a command, PowerShell uses default sort properties defined by the type of the first input object, or compares the objects
themselves.
```

This command displays the files in the current directory by length in ascending order.

```
PS> Get-ChildItem -Path C:\Test -File | Sort-Object 

> ty Length

Author: Are you sure?  It looks like a typing accident.

Reviewer: Thanks. That was an accident.

Reviewer: Fixed

## NOTES

The `Sort-Object` cmdlet sorts objects based on properties specified in the command or the default
sort properties for the object type. Default sort properties are defined using the

> `DefaultKeyPropertySet` element in a `types.ps1xml` file. For more information, see

Author: using the `PropertySet` named `DefaultKeyPropertySet` in a

Reviewer: Fixed it.

[about_Types.ps1xml](/powershell/module/Microsoft.PowerShell.Core/About/about_Types.ps1xml).

If an object does not have one of the specified properties, the property value for that object is
interpreted by `Sort-Object` as **Null** and placed at the end of the sort order.

When no sort properties are available, PowerShell attempts to compare the objects themselves.
`Sort-Object` uses the **Compare** method for each property. If a property does not implement
**IComparable**, the cmdlet converts the property value to a string and uses the **Compare** method
for **System.String**. For more information, see
[PSObject.CompareTo(Object) Method](/dotnet/api/system.management.automation.psobject.compareto).

If you sort on an enumerated property such as **Status**, `Sort-Object` sorts by the enumeration

> values. For Windows services, **Stopped** has a value of **1** and **Running** has a value of **4**.

Author: Are we supposed to say **Windows** rather than **Microsoft Windows**?

Reviewer: **Windows** is sufficient here.
